nle/env/base.py

```diff
@@ -539,9 +551,16 @@ def _perform_known_steps(self, observation, done, exceptions=True):
                 if exceptions:
                     # This causes an annoying unnecessary copy...
                     msg = bytes(observation[self._message_index])
-                    # Allow agent to select stuff to eat, attack, and to
-                    # select directions.
-                    if b"eat" in msg or b"attack" in msg or b"direction?" in msg:
+                    # Do not skip some questions to allow agent to select
+                    # stuff to eat, attack, and to select directions.
+                    decline = True
+                    # I would go with this to avoid surprises when we allow all
+                    # but restrict some.
+                    for el in [b"eat", b"attack", b"direction?", b"pray"]:
```

Comment:
I find this a bit complicated.

We could do something along the lines of `any(el in s for el in ("y", "x", "a"))`?

Also does this do what you want? If `exceptions` is false, we still decline?
Would make sense to refactor this into

```
if not exceptions:
  observation, done = self.env.step(ASCII_ESC)
  continue

# exceptions is True here, so do something else.
```

Reply:
That's a good catch! I fixed the logic. Now when calling step() we either allow all the yn questions or skip those that are not in the exceptions list.